Diagnostics/HealthChecker/Analyzer/Invoke-AnalyzerEngine.ps1

```diff
@@ -399,52 +399,80 @@
+                } else {
+                    $displayWriteType2012 = "Yellow"
+                    $displayValue2012 = "Redistributable is outdated"
+                }
+            }
+
+            if ($detectedVisualRedistVersion.DisplayName -like "Microsoft Visual C++ 2013*") {
+                $vcRedist2013Detected = $true
+                if ($detectedVisualRedistVersion.VersionIdentifier -eq [HealthChecker.VCRedistVersion]::VCRedist2013) {
+                    $displayWriteType2013 = "Green"
+                    $displayValue2013 = "{0} Version is current" -f $detectedVisualRedistVersion.DisplayVersion
+                } else {
+                    $displayWriteType2013 = "Yellow"
+                    $displayValue2013 = "Redistributable is outdated"
+                }
             }
         }
     }
 
+    if (!($vcRedist2012Detected)) {
+        $displayWriteType2012 = "Yellow"
+        $displayValue2012 = "Unknown"
+    }
+
+    if (!($vcRedist2013Detected)) {
+        $displayWriteType2013 = "Yellow"
+        $displayValue2013 = "Unknown"
+    }
+
     $analyzedResults = Add-AnalyzedResultInformation -Name "Visual C++ 2012" -Details $displayValue2012 `
         -DisplayGroupingKey $keyOSInformation `
         -DisplayWriteType $displayWriteType2012 `
         -AnalyzedInformation $analyzedResults
 
-    $analyzedResults = Add-AnalyzedResultInformation -Name "Visual C++ 2013" -Details $displayValue2013 `
-        -DisplayGroupingKey $keyOSInformation `
-        -DisplayWriteType $displayWriteType2013 `
-        -AnalyzedInformation $analyzedResults
-
-    if ($null -ne $osInformation.VcRedistributable -and
-        ($displayWriteType2012 -eq "Yellow" -or
-            $displayWriteType2013 -eq "Yellow")) {
-        $analyzedResults = Add-AnalyzedResultInformation -Details "Note: For more information about the latest C++ Redistributeable please visit: https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads`r`n`t`tThis is not a requirement to upgrade, only a notification to bring to your attention." `
+    if ($exchangeInformation.BuildInformation.ServerRole -ne [HealthChecker.ExchangeServerRole]::Edge) {
+        $analyzedResults = Add-AnalyzedResultInformation -Name "Visual C++ 2013" -Details $displayValue2013 `
             -DisplayGroupingKey $keyOSInformation `
-            -DisplayCustomTabNumber 2 `
-            -DisplayWriteType "Yellow" `
+            -DisplayWriteType $displayWriteType2013 `
             -AnalyzedInformation $analyzedResults
     }
 
+    $vcRedistOutdatedDetails = "Note: For more information about the latest C++ Redistributeable please visit: https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads`r`n`t`tThis is not a requirement to upgrade, only a notification to bring to your attention."
+
+    if ($exchangeInformation.BuildInformation.ServerRole -ne [HealthChecker.ExchangeServerRole]::Edge) {
```

Comment:
We don't need to have this so complicated with multiple `if` statements. We can make it simple and just have one.

If not an Edge server, check for both 2012 and 2013. Otherwise, check for only 2012.

```

if (($exchangeInformation.BuildInformation.ServerRole -ne [HealthChecker.ExchangeServerRole]::Edge -and
	($displayWriteType2012 -eq "Yellow" -or
        $displayWriteType2013 -eq "Yellow")) -or
        $displayWriteType2012 -eq "Yellow") {
	
	$analyzedResults = Add-AnalyzedResultInformation -Details "Note: For more information about the latest C++ Redistributeable please visit: https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads`r`n`t`tThis is not a requirement to upgrade, only a notification to bring to your attention." `
            -DisplayGroupingKey $keyOSInformation `
            -DisplayCustomTabNumber 2 `
            -DisplayWriteType "Yellow" `
            -AnalyzedInformation $analyzedResults
}

```

Reply:
Done.